### DISCUSSION TITLE
How to setup profiling with traces_sampler

### DISCUSSION THREAD

**Discussion by paxw-panevo** [getsentry/sentry-python#3039] [Q&A] [answered]
In the [docs](https://docs.sentry.io/platforms/python/profiling/#enable-profiling), it says that 

> The `profiles_sample_rate` setting is relative to the `traces_sample_rate` setting.

We use a `traces_sampler`. Is my assumption correct that sentry can only profile what's been traced? Is it correct, then, that regardless whether we use `profiles_sampler` or `profiles_sample_rate`, it is still relative to whatever `traces_sampler` returns?

**Comment by antonpirker** [comment, marked as answer]
Yes, you are correct, profiles can be only recorded to sampled traces. 

For example if you have set `profiles_sample_rate=0.2` and in your `traces_sampler` you discard 60% of your traces, there will be profiles attached to 20% (the 0.2) of the remaining 40% of the traces that have been sampled by `traces_sampler`.
